### PR TITLE
Vendor-specific configuration.

### PR DESCRIPTION
**Pull request by cube2222**

### CODE REVIEW

File: spacelift/resource_stack.go [hunk 1]
```diff
@@ -61,6 +61,36 @@ func resourceStack() *schema.Resource {
 				Description: "GitHub branch to apply changes to",
 				Required:    true,
 			},
+			"cloudformation": {
```
Comment (marcinwyszynski) [inline spacelift/resource_stack.go:64]: I think `ConflictsWith` would not hurt here?

File: spacelift/resource_stack.go [hunk 2]
```diff
@@ -109,6 +139,26 @@ func resourceStack() *schema.Resource {
 				Description: "Project root is the optional directory relative to the workspace root containing the entrypoint to the Stack.",
 				Optional:    true,
 			},
+			"pulumi": {
```
Comment (marcinwyszynski) [inline spacelift/resource_stack.go:142]: And here.

File: spacelift/resource_stack.go [hunk 3]
```diff
@@ -316,6 +384,30 @@ func stackInput(d *schema.ResourceData) structs.StackInput {
 		ret.TerraformVersion = toOptionalString(terraformVersion)
 	}
 
+	if cloudFormation, ok := d.Get("cloudformation").([]interface{}); ok && len(cloudFormation) > 0 {
+		ret.VendorConfig = &structs.VendorConfigInput{
+			CloudFormationInput: &structs.CloudFormationInput{
+				EntryTemplateFile: toString(cloudFormation[0].(map[string]interface{})["entry_template_file"]),
+				Region:            toString(cloudFormation[0].(map[string]interface{})["region"]),
+				// TODO: On the backend, make this the stack slug by default.
```
Comment (marcinwyszynski) [inline spacelift/resource_stack.go:392]: Please remove this comment.

Comment (marcinwyszynski) [reply spacelift/resource_stack.go:392]: This is a public repo.

File: spacelift/resource_stack.go [hunk 4]
```diff
@@ -316,6 +384,30 @@ func stackInput(d *schema.ResourceData) structs.StackInput {
...
+				StackName:      toString(cloudFormation[0].(map[string]interface{})["stack_name"]),
+				TemplateBucket: toString(cloudFormation[0].(map[string]interface{})["template_bucket"]),
+			},
+		}
+	} else if pulumi, ok := d.Get("pulumi").([]interface{}); ok && len(pulumi) > 0 {
+		ret.VendorConfig = &structs.VendorConfigInput{
+			Pulumi: &structs.PulumiInput{
+				LoginURL: toString(pulumi[0].(map[string]interface{})["login_url"]),
+				// TODO: On the backend, make this the stack slug by default.
```
Comment (marcinwyszynski) [inline spacelift/resource_stack.go:401]: Same.

File: spacelift/resource_stack.go (hunk 1 above)
Comment (cube2222) [reply spacelift/resource_stack.go:64]: Oh yeah, thanks, forgot about it.

**Comment by cube2222** [comment]
[Add CloudFormation and Pulumi to Terraform Provider.](https://trello.com/c/GdNHfo0L/557-add-cloudformation-and-pulumi-to-terraform-provider)

**Review by marcinwyszynski — APPROVED** [review]

**Pull request merged**